unlock-app/src/hooks/useGetMetadataFor.js

```diff
@@ -0,0 +1,23 @@
+import { useMemo } from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import { signBulkMetadataRequest } from '../actions/keyMetadata'
+
+/**
+ * This hook retrieves metadata for a token
+ * @param {*} address
+ */
+export const useGetMetadataFor = (lockAddress, keyOwner) => {
+  const { account, metadata } = useSelector(state => state)
```

Comment:
I _think_ to test this you'll want to mock `useSelector` so that it returns various values for `metadata` and then check that the hook returns what you expect in those cases.

Additionally, you can check that if `account` is present, `dispatch` gets called with the appropriate message.